mixer/pkg/runtime/config/yaml/converter_test.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint
+//go:generate protoc testdata/all/types.proto -otestdata/all/types.descriptor -I$GOPATH/src/istio.io/istio/vendor/istio.io/api -I.
+//go:generate $GOPATH/src/istio.io/istio/bin/mixer_codegen.sh -f mixer/pkg/runtime/config/yaml/testdata/all/types.proto
+
+package yaml
+
+import (
+	"io/ioutil"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/protoc-gen-gogo/descriptor"
+	yaml "gopkg.in/yaml.v2"
+
+	"istio.io/istio/mixer/pkg/runtime/config/yaml/testdata/all"
+)
+
+var (
+	empty = ``
+
+	unknownField = `
+fxxx: 123
+`
+	simple = `
+str: "mystring"
+i64: 56789
+dbl: 123.456
+b: true
+enm: TWO
+oth:
+  str: "mystring2"
+  i64: 33333
+  dbl: 333.333
+  b: true
+`
+	wantSimpleMsg = &foo.Simple{
```

Comment:
I dont think you need to spell out `wantSimpleMsg`, it can just be `&foo.Simple{}`.
```
wantInst = proto.clone(wantMsg)
proto.Unmarshal([]bytes(simple), wantInst)
```
should give you `wantInst`